gno.land/pkg/gnoweb/url_test.go

```diff
@@ -0,0 +1,130 @@
+package gnoweb
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseGnoURL(t *testing.T) {
+	testCases := []struct {
+		Name     string
+		Input    string
+		Expected *GnoURL
+		Err      *error
+	}{
+		{
+			Name:     "malformed url",
+			Input:    "https://gno.land/r/dem)o:$?",
+			Expected: nil,
+			Err:      &ErrURLMalformedPath,
+		},
+
+		{
+			Name:  "webquery + query",
+			Input: "https://gno.land/r/demo/foo$help&func=Bar&name=Baz",
+			Expected: &GnoURL{
+				Path: "/r/demo/foo",
+				Kind: KindRealm,
+				Args: "",
+				WebQuery: url.Values{
+					"help": []string{""},
+					"func": []string{"Bar"},
+					"name": []string{"Baz"},
+				},
+				Query: url.Values{},
+				Host:  "gno.land",
+			},
+			Err: nil,
+		},
+
+		{
+			Name:  "path args + webquery",
+			Input: "https://gno.land/r/demo/foo:example$tz=Europe/Paris",
+			Expected: &GnoURL{
+				Path: "/r/demo/foo",
+				Kind: KindRealm,
+				Args: "example",
+				WebQuery: url.Values{
+					"tz": []string{"Europe/Paris"},
+				},
+				Query: url.Values{},
+				Host:  "gno.land",
+			},
+			Err: nil,
+		},
+
+		{
+			Name:  "path args + webquery + query",
+			Input: "https://gno.land/r/demo/foo:example$tz=Europe/Paris?hello=42",
+			Expected: &GnoURL{
+				Path: "/r/demo/foo",
+				Kind: KindRealm,
+				Args: "example",
+				WebQuery: url.Values{
+					"tz": []string{"Europe/Paris"},
+				},
+				Query: url.Values{
+					"hello": []string{"42"},
+				},
+				Host: "gno.land",
+			},
+			Err: nil,
+		},
+
+		{
+			Name:  "webquery inside query",
+			Input: "https://gno.land/r/demo/foo:example?value=42$tz=Europe/Paris",
+			Expected: &GnoURL{
+				Path:     "/r/demo/foo",
+				Kind:     KindRealm,
+				Args:     "example",
+				WebQuery: url.Values{},
+				Query: url.Values{
+					"value": []string{"42$tz=Europe/Paris"},
+				},
+				Host: "gno.land",
+			},
+			Err: nil,
+		},
+
+		{
+			Name:  "webquery escaped $",
+			Input: "https://gno.land/r/demo/foo:example%24hello=43$hello=42",
+			Expected: &GnoURL{
+				Path: "/r/demo/foo",
+				Kind: KindRealm,
+				Args: "example$hello=43",
+				WebQuery: url.Values{
+					"hello": []string{"42"},
+				},
+				Query: url.Values{},
+				Host:  "gno.land",
+			},
+			Err: nil,
```

Comment:
Nit: I guess maybe a more intuitive naming would be 
`pkgpath:renderpath?renderquery=123&anotherrenderquery=something`

Not sure though.

Reply:
do you mean just for this specific test right?